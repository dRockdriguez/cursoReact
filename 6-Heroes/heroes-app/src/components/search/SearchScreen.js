import React from "react";
import { useNavigate, useLocation } from "react-router-dom";
import queryString from "query-string";
import { useForm } from "../../hooks/useForm";
import { getHeroeByName } from "../../selectors/getHeroesByName";
import { HeroeCard } from "../heroes/HeroeCard";

export const SearchScreen = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const { q = "" } = queryString.parse(location.search);

  const initialForm = {
    searchText: q,
  };

  const [formValues, handleInputChange] = useForm(initialForm);
  const { searchText } = formValues;

  const heroes = [...getHeroeByName(q)];

  const handleSubmit = (e) => {
    e.preventDefault();
    navigate(`?q=${searchText}`);
  };

  return (
    <>
      <h1>Buscar</h1>

      <hr />

      <div className="row">
        <div className="col-5">
          <form onSubmit={handleSubmit}>
            <input
              type="text"
              placeholder="Buscar heroe"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={searchText}
              onChange={handleInputChange}
            />

            <button type="submit" className="btn btn-outline-primary mt-2">
              Buscar
            </button>
          </form>
        </div>

        <div className="col-7">
          <h4>Resultados</h4>
          <hr />

          {q === "" && (
            <div className="alert alert-info">AÃºn no hay resultados</div>
          )}
          {heroes.map((heroe) => (
            <HeroeCard key={heroe.id} {...heroe} />
          ))}
        </div>
      </div>
    </>
  );
};
