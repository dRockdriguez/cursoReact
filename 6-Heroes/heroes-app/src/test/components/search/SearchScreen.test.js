import { SearchScreen } from "../../../components/search/SearchScreen";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";

describe("SearchScreen", () => {
  test("Debe mostrarse correctamente por defecto", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/serch"]}>
        <SearchScreen></SearchScreen>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".alert-info").text().trim()).toBe("AÃºn no hay resultados");
  });
});
