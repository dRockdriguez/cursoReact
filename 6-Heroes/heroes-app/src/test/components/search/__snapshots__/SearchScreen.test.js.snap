// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen Debe mostrarse correctamente por defecto 1`] = `
Array [
  <h1>
    Buscar
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Buscar heroe"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-primary mt-2"
          type="submit"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        AÃºn no hay resultados
      </div>
    </div>
  </div>,
]
`;
